File: 35-search-insert-position/35-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
         int n= nums.length;
        int start=0;
        int end=n-1;
        while(start<=end){
            int mid=start+(end-start)/2;
            if(target>nums[mid]){
                start= mid+1;
            }
            else if (target<nums[mid]){
                end=mid-1;
            }else{
                return mid;
            }
        }return start;
    }
}

File: 35-search-insert-position/35-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int n = nums.size();
        if(target <= nums[0]){
            return 0;
        }
        if(target > nums[n-1])
            return n;
        int low=0, high = n-1;
        while(low<=high){
            int mid = low + (high-low)/2 ;
            if(nums[mid] == target)
                return mid;
            if(nums[mid] < target)
                low = mid + 1;
            else
                high = mid - 1;
        }
        return low;
    }
};